src/common/FileSystem.cpp

```diff
@@ -1244,7 +1258,71 @@ const std::vector<LoadedPakInfo>& GetLoadedPaks()
 	return loadedPaks;
 }
 
-std::string ReadFile(Str::StringRef path, std::error_code& err)
+// expandSymlink
+// code adapted from DarkPlaces fs.c FS_OpenReadFile
+// commit 1c55fd95ff34d9aafb76c927215ea8fbab5210c6
+// Copyright 2008 divverent
+// GPLv2+
+std::string expandSymlink(Str::StringRef path, Str::StringRef content)
+{
+		const char *filename = path.c_str();
+
+		char linkbuf[MAX_QPATH];
+		strncpy(linkbuf, content.c_str(), MAX_QPATH);
+		int count = strlen(linkbuf);
+
+		const char *mergeslash;
+		char *mergestart;
+
+		// Now combine the paths...
```

Comment:
I don't seem to understand this code... Shouldn't it find the last occurrence of '/' and then do the relative directory handling instead of the first occurrence? I'd really prefer if this was broken out in terms of functions like to make this more readable. With all the pointer operations and magic numbers, it's hard to decipher.

Reply:
I'm not sure to understand all that c magic either (but I'm confident in divverent's work), I would appreciate myself a clean c++ rewrite using c++ strings…

Reply:
this was entirely rewritten